### PR TITLE
Backport fixes

### PR DESCRIPTION
**Pull request by tarekziade**
- Backports visualmetrics.py from sitespeed.io
- Contains a backport of https://bugzilla.mozilla.org/show_bug.cgi?id=1603689
- Fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1608136

### CODE REVIEW

**Comment by tarekziade** [comment]
sitespeed's version biggest change is on the Hero measure. Other stuff look like tweaks.

**Review by gmierz — APPROVED** [review]
Looks good to me, thanks!

File: vendor/visualmetrics.py [hunk 1]
```diff
@@ -1490,7 +1494,21 @@ def calculate_contentful_speed_index(progress, directory):
...
+        logging.debug("Output %s" % output)
+
+        # take the last line of the convert call output
+        lines = [line.strip() for line in output.split("\n") if line.strip()]
+        if len(lines) == 0:
+            logging.debug("Could not find the contentfulness value")
+            return -1
+
+        # extract the value from the last line
+        match = matcher.findall(lines[0])
```
Comment (ncalexan) [inline vendor/visualmetrics.py]: This would be more clear if it was `lines[-1]`, since you say _last_ line.

**Review by ncalexan — APPROVED** [review]

**Review by ncalexan — APPROVED** [review]
Fine by me.

**Review by ncalexan — APPROVED** [review]
It's not clear to me which is the critical piece here: the slightly improved error handling, or the slightly modified cropping/difference calculations?  In any case, I'm happy to track upstream.

File: vendor/visualmetrics.py (hunk 1 above)
Comment (tarekziade) [reply vendor/visualmetrics.py]: we are already in the last line, here we pick the first element in the regexp

**Pull request merged**